### PR TITLE
Css Icons

### LINKED ISSUE #65: Deliver icons as backgrounds for high-traffic situations
Went for a long drive today and may have accidentally come up with a solution allowing for the same flexibility in our icon set that SVG allows, only for PNG.

Proposal

1. Either export from Figma, or build PNGs from our optimized SVG set `@2x` and `#000`.
2. In a `.css` file, export something like `.png-icon.iconLink` with `background-image: url(../path/to/Link.png)`
3. Build atomic coloring classes for `.png-icon` that's based on CSS filters using this [tool](https://codepen.io/sosuke/pen/Pjoqqp). eg. `.png-icon.iconLink.fc-orange-400`. These classes will also have their dark mode equivalents.
4. Provide a PNG helper in the same way we do SVG. eg. `@Png.Link.With("fc-orange-400")`
5. Profit.

Any thoughts before I hack together a proof of concept?

Things to solve / discover:

1. General path issues. How do we keep this self-contained as a straight dependency, and integrate into Stack Overflow
2. Inheritance. If we have `<div class="fc-orange-400"><div class="png-icon"></div></div>`, we'll probably want that icon to be orange.
3. Will just delivering the retina version of these be enough, or will scaling a 2x image on 1x screens be good enough?
4. ???

### PR DESCRIPTION
**Pull request by b-kelly**
Adds support for bundling specific icons as css. This is a formal implementation of two separate implementations already in production (Stacks-Editor icons and SO Teams Achievement icons).

Closes #65 

### Changes

* Add css for adding icons to elements 
  * eg `<span class="svg-icon-bg iconBold"></span>` with optional `native` class
*  Add json file to specify specifically which icons to bundle
  * we don't want to bundle _every_ icon, just a handful of them
* Add build script to automatically transform `cssIcons.json` to css classes
* Add `cssIcons.html` file for previewing (added to `index.html`)

### Follow-up

* Expose in `Stacks`?
* Remove individual implementations in dependent code
* Migrate individual implementations from `icon-bg` to `svg-icon-bg`

### CODE REVIEW

**Comment by aaronshekey** [comment]
> We don't want to bundle _every_ icon, just a handful of them

Shouldn't we though? Wouldn't it be up to consumers to tree shake?

> added to `index.html`

This file is such a crime, but it sure works!

> `<span class="icon-bg iconBold"></span>` with optional `native` class

What if we did `svg-icon svg-icon__bg`. Would that cause too many conflicts? If so, I think `svg-icon-bg` could be a consistent way of naming these.

**Comment by b-kelly** [comment]
@aaronshekey You can't tree shake css, unfortunately, unless we put every icon class in it's own file and made users import them one at a time into their existing preprocessors.

I can rename the classes without few issues, but how do you envision an element looking with those classes? Something like:

```html
<span class="svg-icon svg-icon__bg iconFaceMindBlown native"></span>
```

Imo, that's a little wordy. I like `svg-icon-bg` a little better:

```html
<span class="svg-icon-bg iconFaceMindBlown native"></span>
```

File: build/icons.css [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    width: 21px;
+}
+
+.icon-bg.iconOrderedList {
+    --bg-icon: url("data:image/svg+xml,%3csvg width='18' height='18' viewBox='0 0 18 18' fill='none' xmlns='http://www.w3.org/2000/svg'%3e %3cpath fill-rule='evenodd' clip-rule='evenodd' d='M3 6H4V2H2V3H3V6ZM3.8 8H2V7H5V7.9L3.2 10H5V11H2V10.1L3.8 8ZM2 13V12H5V16H2V15H4V14.5H3V13.5H4V13H2ZM7 5V3H16V5H7ZM7 15H16V13H7V15ZM16 10H7V8H16V10Z' fill='black'/%3e %3c/svg%3e");
+}
+
+.icon-bg.iconPencilSm {
+    --bg-icon: url("data:image/svg+xml,%3csvg width='14' height='14' viewBox='0 0 14 14' fill='none' xmlns='http://www.w3.org/2000/svg'%3e %3cpath d='M11.0937 1.71081L12.2347 2.83019C12.4347 3.03005 12.4347 3.33983 12.2347 3.53968L11.0898 4.70381L9.21 2.86474L10.3837 1.71081C10.5837 1.51095 10.8937 1.51095 11.0937 1.71081Z' fill='black'/%3e %3cpath d='M2 10.12L8.37 3.69427L10.2466 5.57034L3.88 12H2V10.12Z' fill='black'/%3e %3c/svg%3e");
+    width: 14px; height: 14px;
```
Comment (aaronshekey) [inline build/icons.css]: Can you give me, an actual maniac, a new line between these properties?

**Comment by aaronshekey** [comment]
> You can't tree shake css

Sure sure, maybe not by that name, but it's possible to run `purgecss` or something similar. 🤔 

```
<span class="svg-icon-bg iconFaceMindBlown native"></span>
```

This works!

Thanks for pulling this together!

**Review by abovedave — APPROVED** [review]
LGTM. Only extra credit would be an update to the readme to describe usage, how to add icons with the json etc

File: build/icons.css (hunk 1 above)
Comment (b-kelly) [reply build/icons.css]: I could, but changing the ` ` to `\n` in the json gives me this:

```css
.icon-bg.iconPencilSm {
    --bg-icon: url("data:image/svg+xml,...");
    width: 14px;
height: 14px;
}
```

and I'm too lazy to add code to fix it.

> Contributions accepted :trollface:

**Comment by b-kelly** [comment]
Updated to `svg-icon-bg` and added README content

**Pull request merged**